napari/_qt/layers/qt_image_base_layer.py

```diff
@@ -56,13 +59,83 @@ def __init__(self, layer):
     def changeColor(self, text):
         self.layer.colormap = text
 
-    def _on_clim_change(self, event=None):
+    def _clim_mousepress(self, event):
+        """when right clicking the clim slider, we want a QHRangeSliderPopup
+        to appear that provides greater control over the values and range of
+        the contrast limits.
+        """
+        if event.button() == Qt.RightButton:
+            precision = 2
+            try:
+                layer_is_int = np.issubdtype(self.layer.data.dtype, np.integer)
+                precision = 0 if layer_is_int else precision
+            except AttributeError:
+                layer_is_int = False
+
+            self.clim_pop = QRangeSliderPopup(
+                initial_values=self.layer.contrast_limits,
+                data_range=self.layer.contrast_limits_range,
+                collapsible=False,
+                precision=precision,
+                parent=self,
+            )
+
+            set_clim = partial(setattr, self.layer, 'contrast_limits')
+            set_crange = partial(setattr, self.layer, 'contrast_limits_range')
+            self.clim_pop.slider.valuesChanged.connect(set_clim)
+            self.clim_pop.slider.rangeChanged.connect(set_crange)
+
+            def reset():
+                self.layer.reset_contrast_limits()
+                self.layer.contrast_limits_range = self.layer.contrast_limits
+
+            btn = QPushButton("reset")
+            btn.setObjectName("reset_clims_button")
+            btn.setToolTip("autoscale contrast to data range")
+            btn.setFixedWidth(40)
+            btn.clicked.connect(reset)
+            self.clim_pop.layout.addWidget(btn)
+            # the "full range" button doesn't do anything if it's not an
+            # unsigned integer type (it's unclear what range should be set)
+            # surface layers will also not have a dtype
+            if np.issubdtype(self.layer.dtype, np.unsignedinteger):
+
+                def reset_range():
+                    self.layer.reset_contrast_limits_range()
+
+                btn = QPushButton("full range")
+                btn.setObjectName("full_clim_range_button")
+                btn.setToolTip("set contrast range to full bit-depth")
+                btn.setFixedWidth(65)
+                btn.clicked.connect(reset_range)
+                self.clim_pop.layout.addWidget(btn)
+                self.clim_pop.finished.connect(self.clim_pop.deleteLater)
+
+            self.clim_pop.show_at('top')
+            self.clim_pop.curmin_label.update_position()
+            self.clim_pop.curmax_label.update_position()
```

Comment:
thanks, broken up and applied the full range button to all integer types.  I made them as standalone functions rather than methods, just in case we want to use them elsewhere with "rangey" type data... but I left them in the same module for now.  sound ok?